### PR TITLE
SOLR-16298: Clean javadocs directory before building

### PR DESCRIPTION
**Pull request by HoustonPutman**
https://issues.apache.org/jira/browse/SOLR-16298

Steps to reproduce the issue:

1. Check out the 9.0.0 tag
2. `./gradlew documentation`
3. Check out main
4. `./gradlew documentation`
5. `less solr/documentation/build/site/core/org/apache/solr/handler/admin/CoreAdminHandlerApi.html`

Notice that the Solr version on this file will read 9.0.0, not 10.0.0-SNAPSHOT.

### CODE REVIEW

File: gradle/documentation/render-javadoc.gradle [hunk 1]
```diff
@@ -268,6 +268,13 @@ class RenderJavadocTask extends DefaultTask {
 
     // create the directory, so relative link calculation knows that it's a directory:
     outputDir.mkdirs();
+    outputDir.listFiles().each {childFile ->
```
Comment (madrob) [inline gradle/documentation/render-javadoc.gradle:271]: what about `project.delete`?

Comment (uschindler) [reply gradle/documentation/render-javadoc.gradle:271]: No idea why it is this way. Have you checked the Lucene version, it may be more up-to-date.

Comment (uschindler) [reply gradle/documentation/render-javadoc.gradle:271]: I agree. This is more Gradle like. You should also check the Lucene version. Maybe it is more up to date in that regard. The version number is an input property there.

Comment (HoustonPutman) [reply gradle/documentation/render-javadoc.gradle]: I looked through the lucene implementation but I didn't really see this addressed either. Maybe I missed it though.

Will change to use `project.delete` though.

**Review by madrob — APPROVED** [review]
There's probably some magic going on about incremental tasks that makes this not do what we think it should do, but current approach seems fine.

**Pull request merged**